cpp/src/shuffler/shuffler.cpp

```diff
@@ -325,36 +336,36 @@ class Shuffler::Progress {
         // requested data.
         {
             auto const t0_init_gpu_data_send = Clock::now();
-            RAPIDSMPF_NVTX_SCOPED_RANGE("init_gpu_send");
-            int i = 0;
-            while (true) {
-                auto const [msg, src] = shuffler_.comm_->recv_any(ready_for_data_tag);
-                if (msg) {
-                    auto ready_for_data_msg = ReadyForDataMessage::unpack(msg);
-                    auto chunk = extract_value(outgoing_chunks_, ready_for_data_msg.cid);
-                    log.trace(
-                        "recv_any from ",
-                        src,
-                        ": ",
-                        ready_for_data_msg,
-                        ", sending: ",
-                        chunk
+            RAPIDSMPF_NVTX_SCOPED_RANGE(
+                "init_gpu_send",
+                std::transform_reduce(
+                    ready_ack_receives_.begin(),
+                    ready_ack_receives_.end(),
+                    0,
+                    std::plus<>(),
+                    [](auto& kv) { return kv.second.size(); }
+                )
+            );
+            for (auto& [dst, futures] : ready_ack_receives_) {
+                auto finished = shuffler_.comm_->test_some(futures);
```

Comment:
Maybe we should add a comment explaining why we need to loop over destinations individually and not the whole `ready_ack_receives_` at once. This is the type of thing I can see ourselves optimizing away in the future.